### PR TITLE
frontend-plugin-api: add IconsApi

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

This adds an `IconsApi` as well as a an `icons` option to `createApp`. These are stop-gap solutions to make sure we have feature parity with the existing frontend system.

This also expands on the compatibility wrapper so that it use the `IconsApi` and `ComponentsApi` in the implementation of the old `AppContext`.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/frontend-plugin-api/src/apis/definitions/IconsApi.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * API for accessing app icons.
+ *
+ * @public
+ */
+export interface IconsApi {
+  getIcon(key: string): IconComponent | undefined;
+
+  listIconKeys(): { keys: Iterable<string> };
```
Comment (freben) [inline packages/frontend-plugin-api/src/apis/definitions/IconsApi.ts:28]: Iterable isn't particularly convenient. Would be worth just making an array, I'd bet.

Also, while we do like object form results, the method name is VERY specific, so may just consider making it string[]

File: packages/frontend-app-api/src/apis/implementations/IconsApi/IconsApi.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  constructor(icons: { [key in string]: IconComponent }) {
+    this.#icons = new Map(Object.entries(icons));
+  }
+
+  getIcon(key: string): IconComponent | undefined {
+    return this.#icons.get(key);
+  }
+
+  listIconKeys: IconsApi['listIconKeys'] = () => {
```
Comment (freben) [inline packages/frontend-app-api/src/apis/implementations/IconsApi/IconsApi.ts:35]: Why instantiate this one but not the other?

**Review by freben — APPROVED** [review]
Alright, looking good but wondering about the iterator

File: packages/frontend-plugin-api/src/apis/definitions/IconsApi.ts (hunk 1 above)
Comment (Rugvip) [reply packages/frontend-plugin-api/src/apis/definitions/IconsApi.ts:28]: Yeah it's not intended to be called particularly often after all.

Happy to make it a simple return, another option I considered was more of a `getIconsMeta(): { ... }`. Didn't wanna get stuck for too long though

File: packages/frontend-app-api/src/apis/implementations/IconsApi/IconsApi.ts (hunk 2 above)
Comment (Rugvip) [reply packages/frontend-app-api/src/apis/implementations/IconsApi/IconsApi.ts:35]: it was the super complex signature, got rid of it now though with just returning an array

File: packages/frontend-plugin-api/src/apis/definitions/IconsApi.ts (hunk 1 above)
Comment (Rugvip) [reply packages/frontend-plugin-api/src/apis/definitions/IconsApi.ts]: Switched to `(): string[]`

**Pull request merged**